ACE/ace/Functor.inl

```diff
@@ -104,6 +104,22 @@ ACE_Hash<unsigned long>::operator () (unsigned long t) const
   return t;
 }
 
+#if (ACE_SIZEOF_LONG == 8)
```

Comment:
Shouldn't this also be size_t? The check for the sizeof looks also strang

Reply:
The check is to avoid redefining hash for `long long` (and `unsigned long long`) twice. If there was no check and size of `long` < 8 bytes, then `ACE_INT64` will be `long long` when C++11 is not enabled ( in case C++11 is enabled, `int64_t` is used and in many platforms it's implemented as `long long` as well) and the hash for `long long` will be defined twice.